heareval/tasks/nsynth_pitch.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+"""
+Pre-processing pipeline for NSynth pitch detection
+"""
+
+import os
+from pathlib import Path
+from functools import partial
+import logging
+
+import luigi
+import pandas as pd
+from slugify import slugify
+
+import heareval.tasks.config.nsynth_pitch as config
+import heareval.tasks.util.luigi as luigi_util
+
+logger = logging.getLogger("luigi-interface")
+
+# Set the task name for all WorkTasks
+luigi_util.WorkTask.task_name = config.TASKNAME
+
+
+class ExtractArchiveTrain(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.TRAIN_DOWNLOAD_URL, outfile="train-corpus.tar.gz"
+            )
+        }
+
+
+class ExtractArchiveValidation(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.VALIDATION_DOWNLOAD_URL, outfile="valid-corpus.tar.gz"
+            )
+        }
+
+
+class ExtractArchiveTest(luigi_util.ExtractArchive):
+    def requires(self):
+        return {
+            "download": luigi_util.DownloadCorpus(
+                url=config.TEST_DOWNLOAD_URL, outfile="test-corpus.tar.gz"
+            )
+        }
+
+
+class ConfigureProcessMetaData(luigi_util.WorkTask):
+    """
+    This config is data dependent and has to be set for each data
+    """
+
+    outfile = luigi.Parameter()
+
+    def requires(self):
+        return {
+            "train": ExtractArchiveTrain(infile="train-corpus.tar.gz"),
+            "valid": ExtractArchiveValidation(infile="valid-corpus.tar.gz"),
+            "test": ExtractArchiveTest(infile="test-corpus.tar.gz"),
+        }
+
```

Comment:
These next two methods seem generic? But maybe it makes sense to keep them here?

Reply:
Ya, they are different than google speech though - so I think it makes sense to leave it

Reply:
@jorshi maybe because GSC has like postprocessing? that seems like it would be a common pattern